app/controllers/avatars_controller.rb

```diff
@@ -0,0 +1,27 @@
+class AvatarsController < ApplicationController
+  before_action :authenticate_user!
+
+  def upload
+    @avatar = Avatar.find_by(user_id: current_user.id) || Avatar.new
+    file = params[:avatar][:file]
+    @saved = @avatar.update(
+      file: file.read,
+      filename: file.original_filename,
+      filesize: file.size,
+      mime_type: file.content_type,
+      user_id: current_user.id
```

Comment:
There shouldn't be a `params[:user_id]` because avatars are not a subresource of the user. But because one should only change the profile picture of oneself, `current_user` is also the desired behaviour here.